src/features/accuse/index.tsx

```diff
@@ -0,0 +1,142 @@
+    setSarcophagusId(e.target.value);
+  }
+
+  function handleAddPrivateKey() {
+    setArchaeologistPrivateKeys(prev => [...prev, '']);
+  }
+
+  function handleRemovePrivateKey(index: number) {
+    const newArchaeologistPrivateKeys = [...archaeologistPrivateKeys];
+    newArchaeologistPrivateKeys.splice(index, 1);
+
+    // if there are no archaeologist private keys left, add an empty one
+    if (newArchaeologistPrivateKeys.length === 0) {
+      newArchaeologistPrivateKeys.push('');
+    }
+
+    setArchaeologistPrivateKeys(newArchaeologistPrivateKeys);
+  }
+
+  function handleChangePrivateKey(e: React.ChangeEvent<HTMLTextAreaElement>, index: number) {
+    const newArchaeologistPrivateKeys = [...archaeologistPrivateKeys];
+    newArchaeologistPrivateKeys[index] = e.target.value;
+    setArchaeologistPrivateKeys(newArchaeologistPrivateKeys);
+  }
+
+  function handleChangePaymentAddress(e: React.ChangeEvent<HTMLInputElement>) {
+    setPaymentAddressOverride(e.target.value);
+  }
+
+  function handleAccuse() {
+    accuse();
+  }
+
+  // Reset the form when the accuse transaction is successful
+  useEffect(() => {
+    if (isSuccess) {
+      setSarcophagusId('');
+      setArchaeologistPrivateKeys(['']);
+      setPaymentAddressOverride('');
+    }
+  }, [isSuccess, setArchaeologistPrivateKeys]);
+
+  return (
+    <Flex
+      w="100%"
+      direction="column"
+      pr={50}
+    >
+      <Text fontSize="2xl">Accusals</Text>
+      <Text
+        mt={3}
+        variant="secondary"
+      >
+        If you found a keyshare an archaeologist has leaked, you can accuse them on this page.
+        Please enter the Sarcophagus ID and the Archaeologogist keyshare as evidence.
+      </Text>
+      <FormControl>
+        <FormLabel mt={12}>Sarcophagus ID</FormLabel>
+        <Input
+          placeholder="0x000..."
+          value={sarcophagusId}
+          onChange={handleChangeSarcophagusId}
+          mt={1}
+          p={6}
+        />
+        <FormLabel
+          mt={12}
+          mb={1}
+        >
+          Archaeologist Private Key
+        </FormLabel>
+        <>
+          {archaeologistPrivateKeys.map((privateKey, index) => (
+            <PrivateKeyInput
+              key={index}
+              value={privateKey}
+              onChange={e => handleChangePrivateKey(e, index)}
+              onClickAddButton={handleAddPrivateKey}
+              onClickRemoveButton={() => handleRemovePrivateKey(index)}
+              showAddButton={index === archaeologistPrivateKeys.length - 1}
+              hideRemoveButton={
+                archaeologistPrivateKeys.length === 1 && archaeologistPrivateKeys[0].trim() === ''
+              }
+            />
+          ))}
+        </>
+        <Text mt={3}>
+          To accuse multiple achaeologists, click the plus sign to the right of the field. You must
+          accuse all together if you plan to accuse multiple.
```

Comment:
what does "you must accuse all together" mean, outside of supplying private keys for each one?

Reply:
This statement seems a little unnecessary now that I'm looking at it. Simply saying "To accuse multiple achaeologists, click the plus sign to the right of the field" should be enough right?